__tests__/commands/_helpers.js

```diff
@@ -79,13 +79,24 @@ export async function run<T, R>(
 
   const reporter = new Reporter({stdout, stderr: stdout});
 
-  const dir = path.join(fixturesLoc, name);
-  const cwd = path.join(
-    os.tmpdir(),
-    `yarn-${path.basename(dir)}-${Math.random()}`,
-  );
-  await fs.unlink(cwd);
-  await fs.copy(dir, cwd, reporter);
+  let cwd;
+  if (fixturesLoc) {
+    // if fixture loc is not set create a tmp dir so that we don't copy CWD during test run
+    const dir = path.join(fixturesLoc, name);
+    cwd = path.join(
+      os.tmpdir(),
+      `yarn-${path.basename(dir)}-${Math.random()}`,
+    );
+    await fs.unlink(cwd);
+    await fs.copy(dir, cwd, reporter);
+  } else {
+    cwd = path.join(
+      os.tmpdir(),
+      `yarn-${Math.random()}`,
```

Comment:
Just found that it does support it natively in Nodejs 5.10+: https://nodejs.org/api/fs.html#fs_fs_mkdtemp_prefix_options_callback